site/app/controllers/admin/LateController.php

```diff
@@ -280,6 +281,29 @@ public function updateExtension() {
         }
     }
 
+    /**
+     * @AccessControl(role="INSTRUCTOR")
+     * @Route("/courses/{_semester}/{_course}/users/view_latedays", methods={"GET"})
+     * @return RedirectResponse|WebResponse
+     **/
+    public function viewStudentLatedays() {
+        if (!isset($_GET["student_id"])) {
+            $this->core->addErrorMessage("No student ID provided");
+            return new RedirectResponse($this->core->buildCourseUrl(['users']));
+        }
+        $student_id = $_GET["student_id"];
+        $user = $this->core->getQueries()->getUserById($student_id);
+        if ($user === null) {
+            $this->core->addErrorMessage("Invalid Student ID \"" . $_POST["student_id"] . "\"");
+            return new RedirectResponse($this->core->buildCourseUrl(['users']));
+        }
+        return new WebResponse(
+            ['LateDaysTable'],
+            'showLateTabletoInstructor',
+            LateDays::fromUser($this->core, $this->core->getQueries()->getUserById($student_id))
```

Comment:
You already have a variable `$user` defined.  Use that instead of calling `$this->core->getQueries()->getUserById()` again.

Reply:
Made adequate changes.